src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -427,11 +429,36 @@ class MavlinkStreamStatustext : public MavlinkStream
 			if (_mavlink->get_logbuffer()->get(&mavlink_log)) {
 
 				mavlink_statustext_t msg;
+				const char *text = mavlink_log.text;
+				constexpr unsigned max_chunk_size = sizeof(msg.text);
 				msg.severity = mavlink_log.severity;
-				strncpy(msg.text, (const char *)mavlink_log.text, sizeof(msg.text));
-				msg.text[sizeof(msg.text) - 1] = '\0';
+				msg.chunk_seq = 0;
+				msg.id = id++;
+				unsigned text_size;
+
+				while ((text_size = strlen(text)) > 0) {
+					unsigned chunk_size = math::min(text_size, max_chunk_size);
+
+					if (chunk_size < max_chunk_size) {
+						memcpy(&msg.text[0], &text[0], chunk_size + 1);
```

Comment:
What's the `+ 1` for?

Reply:
It used to be for terminating the string, but you are right ... the memset does the job as well.